### PR TITLE
[CARBONDATA-1101] Avoid widening between wrapper classes

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
Modification reason:
Widening between wrapper classes throws ClassCastException

### CODE REVIEW

**Comment by kumarvishal09** [comment]
LGTM

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
Can one of the admins verify this patch?

File: processing/src/main/java/org/apache/carbondata/processing/newflow/sort/unsafe/merger/UnsafeIntermediateFileMerger.java [hunk 1]
```diff
@@ -314,7 +314,7 @@ private void writeDataTofile(Object[] row) throws CarbonSortKeyAndGroupByExcepti
           case SHORT:
           case INT:
           case LONG:
-            rowData.putLong(size, (Long) value);
+            rowData.putLong(size, Long.valueOf(value.toString()));
```
Comment (ravipesala) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/sort/unsafe/merger/UnsafeIntermediateFileMerger.java]: Should be kept in the respective datatypes, it is already fixed.

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2153/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-2.1/47/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/18/

**Comment by mohammadshahidkhan** [comment]
this closed with  #979

**Pull request closed without merging**